### PR TITLE
Cypress core app workflows

### PR DESCRIPTION
**Pull request by ashblue**
We now have Cypress coverage for most of our critical pathways.

image

## What This Covers

* Basic terminal loading and closing
* Terminal workflow from approval to initial request to displaying the terminal
* Open-source login workflows
* Dev login
* Most settings CRUD

## What this doesn't cover

I think it's a good idea to cover these features when time permits. Before adding these we should let the current suite of E2Es stabilize in case we notice any bugs.

* Global activity and downloading the audit log
* Approvals
  * Cover everything outside of the basic approved workflow
* Terminals
  * Advanced terminal filtering commands
  * Share links
  * search, uploads, downloads
  * Approval "Reason Required" workflow
  * Changes in terminal access activity
  * Record output terminal option
* Settings
  * Tweaking a number of the approval settings
  * GitHub connection workflow (can we test this?)
  * Change user access to prompts in the user's settings page

## Known Issues

* Runbooks vault is erroring when trying to adjust runbook settings. Separate ticket to re-enable the suite there https://linear.app/cased/issue/CAS-477/[helm]-configure-api-vault-connection
* Approval workflow websocket is not working in the cloud as expected https://linear.app/cased/issue/CAS-478/terminal-approval-window-timing-out

## Long-Term Recommendations

* Setup Xterm.js to print out the last 100 statements to a hidden DOM `<div>` in Cypress mode only. Xterm is in Canvas so it's impossible to actually read the output right now. `wait()` hacks being used instead atm which isn't good.
* Move most selectors over to test IDs for sustainability reasons. Currently using direct selectors which isn't great
* Move environment population to dev-only API commands so testable content can be generated on demand in Cypress

### CODE REVIEW

**Comment by ashblue** [comment]
@jnewland this works great. Ran through it and fiddled with tweaking a test (no problems). I'll leave this open as a draft while the PR to tie everything up with the CI goes through.

**Comment by ashblue** [comment]
@jnewland am I formatting this file right to add in the missing prompts? Have the sneaking suspicion my syntax is incorrect https://github.com/cased/shell/pull/69/commits/75caf92f13d2c4add64e37bb3454a1096001a041

**Comment by jnewland** [comment]
@ashblue I pushed up a couple commits that added an example, but it looks like we may need to add or adjust a few Prompts to make all of the tests pass. This is all made harder because of the divergence between dev and CI: the development environment's list of prompts is loaded from [`tests/data/jump-local-run-docker.json`](https://github.com/cased/shell/blob/ash/cas-215-implement-cypress-for-core-workflows/tests/data/jump-local-run-docker.json) and the list in CI comes from [`.devcontainer/helm/shell-dev.yaml`](https://github.com/cased/shell/blob/ash/cas-215-implement-cypress-for-core-workflows/.devcontainer/helm/shell-dev.yaml). In the long term it'd be nice to standardize those somehow, of course! For now, feel free to adjust either of them to fit your needs.

**Comment by ashblue** [comment]
@jnewland I probably need to adjust some of the pre-existing tests still. I'll take a look today and let you know what I find out.

File: apps/shell-e2e/src/e2e/settings/settings-runbooks.cy.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import { nanoid } from 'nanoid';
+
+describe('settings - runbooks', () => {
+  beforeEach(() => {
+    cy.loginAdmin();
+    cy.contains('a', 'Settings').click();
+    cy.contains('.tabs a', 'Runbooks').click();
+  });
+
+  xdescribe('API Providers', () => {
```
Comment (rikasun) [inline apps/shell-e2e/src/e2e/settings/settings-runbooks.cy.ts:10]: Did you mean to add x here?

**Review by rikasun — APPROVED** [review]
🥳🥳🥳

**Review by jnewland — APPROVED** [review]
Awesome work @ashblue 🎉

**Pull request merged**